src/plugin/file/kml/kmlparser.js

```diff
@@ -841,6 +849,55 @@ plugin.file.kml.KMLParser.prototype.createTreeNode_ = function(el, opt_parent) {
 };
 
 
+/**
+ * Examine styles that are unsupported in OpenLayers KML styles.
+ * @param {Node} node Node.
+ * @private
+ */
+plugin.file.kml.KMLParser.prototype.examineStyles_ = function(node) {
+  var n;
+  for (n = node.firstElementChild; n; n = n.nextElementSibling) {
+    if (n.localName == 'BalloonStyle') {
+      var properties = ol.xml.pushParseAndPop({}, plugin.file.kml.BALLOON_PROPERTY_PARSERS, n, []);
+      if (properties) {
+        var id = node.id || node.getAttribute('id');
+        this.otherStyleMap[id] = properties;
+      }
+    }
+  }
+};
+
+
+/**
+ * Read the KML balloon style.
+ * @param {ol.Feature} feature The feature
+ * @private
+ */
+plugin.file.kml.KMLParser.prototype.readBalloonStyle_ = function(feature) {
+  var styleUrl = /** @type {string} */ (feature.get('styleUrl'));
+  var style = this.findStyle_(decodeURIComponent(styleUrl), false, true);
+  var text = style.text;
+  var bgColor = style.bgColor || '255, 255, 255, 1';
+  var textColor = style.textColor || '0, 0, 0, 1';
+
+  if (style && text) {
+    var pattern = /[$]\[(.*?)\]/g;
+    var regex = new RegExp(pattern);
+
+    text = text.replace(regex, function(match) {
+      var key = match.slice(2, -1);
+      if (key in feature.values_) {
+        return feature.get(key);
+      } else {
+        return '';
+      }
+    });
+    var description = '<div style="background:rgba(' + bgColor + ');color:rgba(' + textColor + ')">' + text + '</div>';
+    feature.set('description', description);
```

Comment:
Do not store this in the `description` property on the feature. That field is already used for the KML `<description>` tag on the Placemark, and we don't want to wipe that out.

Reply:
I told him to do that. What's the point of keeping it around if we only want to show the other?

Reply:
While they may contain the same content, that's up to the KML and it's possible they'll each have unique content. We don't want to discard the description in favor of the balloon text so we need a way to show both.

Google Earth shows the balloon text on click and the description in the Get Info context menu option. I'll look into getting a balloon implementation we can use to be closer to that behavior.